eo-maven-plugin/src/main/java/org/eolang/maven/ProbeMojo.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2022 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import com.jcabi.log.Logger;
+import com.jcabi.xml.XML;
+import com.jcabi.xml.XMLDocument;
+import com.yegor256.tojos.Tojo;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.TreeSet;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.cactoos.iterable.Filtered;
+import org.cactoos.list.ListOf;
+import org.eolang.maven.hash.ChNarrow;
+import org.eolang.maven.hash.ChResolve;
+import org.eolang.maven.hash.CommitHash;
+import org.eolang.maven.objectionary.Objectionary;
+import org.eolang.maven.objectionary.OyFallbackSwap;
+import org.eolang.maven.objectionary.OyHome;
+import org.eolang.maven.objectionary.OyRemote;
+import org.eolang.maven.util.Online;
+import org.eolang.maven.util.Rel;
+
+/**
+ * Go through all `probe` metas in XMIR files, try to locate the
+ * objects pointed by `probe` in Objectionary and if found register them in
+ * catalog.
+ *
+ * @since 0.28.11
+ * @checkstyle CyclomaticComplexityCheck (300 lines)
+ * @todo #1395:30min Need to add the logic of "hasReservedChars" method to
```

Comment:
@MikhailLipanin I believe that the better place for that `@todo` right above the `hasReservedChars` method. What do you think?